app/models/spree_gateway/payment_decorator.rb

```diff
@@ -1,8 +1,60 @@
 module SpreeGateway
   module PaymentDecorator
+    def self.prepended(base)
+      # Added the 'intent' state to allow payment gateway to handle creation of payment intent.
+      # Overridden here for now, but assume this would be better sitting in core.
+      base.state_machine initial: :checkout do
```

Comment:
I'm wondering if we need this additional state related to intent, or if we could just save a `Payment` object for StripeElementsGateway only if the payment was successfully created?

Reply:
But if that needs to happen, this would probably be better of in core. This will be a breaking change though for other payment gateways.

Reply:
@rafalcymerys Yes, I was unsure if there was a way to avoid adding the extra 'intent' payment state. In the end I felt like it was significant enough to warrant a separate state, and seemed to make sense as a way to call the Stripe Gateway. If you were to remove the 'intent' state, how would you suggest calling the Stripe to create the payment intent? Would it be some sort of before_action related to payment? And would it go via the spree_gateway?